external-import/socradar/src/lib/radar.py

```diff
@@ -0,0 +1,354 @@
+                if self._matches_pattern(value, "md5"):
+                    return File(
+                        id=f"file--{str(uuid.uuid4())}",
+                        type="file",
+                        hashes={"MD5": value},
+                    )
+                elif self._matches_pattern(value, "sha1"):
+                    return File(
+                        id=f"file--{str(uuid.uuid4())}",
+                        type="file",
+                        hashes={"SHA-1": value},
+                    )
+                elif self._matches_pattern(value, "sha256"):
+                    return File(
+                        id=f"file--{str(uuid.uuid4())}",
+                        type="file",
+                        hashes={"SHA-256": value},
+                    )
+            return None
+        except Exception as e:
+            self.helper.log_error(f"Error creating observable for {value}: {str(e)}")
+            return None
+
+    def _matches_pattern(self, value, pattern_name):
+        """Step 3.2: Check if value matches a regex pattern"""
+        import re
+
+        return re.match(self.regex_patterns[pattern_name], value) is not None
+
+    def _create_stix_pattern(self, value, feed_type):
+        """Create STIX pattern based on feed type"""
+        # Direct mapping for feed types
+        if feed_type == "url":
+            return f"[url:value = '{value}']"
+        elif feed_type == "domain":
+            return f"[domain-name:value = '{value}']"
+        elif feed_type == "ip":  # Add handling for IP type
+            if re.match(self.regex_patterns["ipv4"], value):
+                return f"[ipv4-addr:value = '{value}']"
+            elif re.match(self.regex_patterns["ipv6"], value):
+                return f"[ipv6-addr:value = '{value}']"
+        elif feed_type == "hash":
+            if re.match(self.regex_patterns["md5"], value):
+                return f"[file:hashes.'MD5' = '{value}']"
+            elif re.match(self.regex_patterns["sha1"], value):
+                return f"[file:hashes.'SHA-1' = '{value}']"
+            elif re.match(self.regex_patterns["sha256"], value):
+                return f"[file:hashes.'SHA-256' = '{value}']"
+
+        # Fallback to pattern detection if feed_type doesn't match
+        for pattern_type, regex in self.regex_patterns.items():
+            if re.match(regex, value):
+                if pattern_type == "url":
+                    return f"[url:value = '{value}']"
+                elif pattern_type == "domain":
+                    return f"[domain-name:value = '{value}']"
+                elif pattern_type == "ipv4":
+                    return f"[ipv4-addr:value = '{value}']"
+                elif pattern_type == "ipv6":
+                    return f"[ipv6-addr:value = '{value}']"
+
+        self.helper.log_error(
+            f"Could not determine pattern for value: {value} (type: {feed_type})"
+        )
+        return None
+
+    def run(self):
```

Comment:
It would be easier to use Opencti Scheduler

Here is an example of usage : https://github.com/OpenCTI-Platform/connectors/blob/5a346b3902308712efa2ef4c9f7069fa3f6389be/templates/external-import/src/external_import_connector/connector.py#L189